src/server/server_state.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/server_state.h"
+
+#include "base/logging.h"
+#include "server/conn_context.h"
+
+namespace dfly {
+
+// thread_id(util::ProactorBase::GetIndex()
+
+MonitorsRepo::MonitorInfo::MonitorInfo(ConnectionContext* conn) : connection(conn) {
+}
+
+void MonitorsRepo::MonitorInfo::Send(std::string msg) {
+  CHECK(connection);
+  connection->SendMonitorMsg(msg);
+}
+
+void MonitorsRepo::Add(const MonitorInfo& info) {
+  VLOG(1) << "register connection "
+          << " at address 0x" << std::hex << (const void*)info.connection << " for thread "
+          << util::ProactorBase::GetIndex();
+
+  monitors_.push_back(info);
+}
+
+void MonitorsRepo::Send(std::string msg) {
+  if (!monitors_.empty()) {
+    VLOG(1) << "thread " << util::ProactorBase::GetIndex() << " sending monitor message '" << msg
+            << "' for " << monitors_.size();
+    std::for_each(monitors_.begin(), monitors_.end(),
+                  [msg](auto& monitor_conn) { monitor_conn.Send(msg); });
```

Comment:
use `[&]` capture (i personally prefer a simple loop that will be 1 line longer but more readable without lambdas).

Reply:
changed to loop